src/routes/promises/+page.server.ts

```diff
@@ -36,26 +42,36 @@ export interface PromisesByCategory {
 	count: number;
 }
 
+export type AssemblyMember = ReturnType<typeof getAssemblyMembers>[number];
+
 const CURRENT_CABINET_ASSEMBLY_ID = 'คณะรัฐมนตรี-64';
+const PREVIOUS_CABINET_ASSEMBLY_ID = 'คณะรัฐมนตรี-63';
 
-export async function load() {
-	const cabinetAssembly = (await fetchAssemblies()).find(
-		(a) => a.id === CURRENT_CABINET_ASSEMBLY_ID
-	);
+function findCabinetAssemblyWithId(assemblies: Assembly[], id: string) {
+	const cabinetAssembly = assemblies.find((a) => a.id === id);
 	if (!cabinetAssembly) {
-		error(500, `Cannot find the current cabinet: ${CURRENT_CABINET_ASSEMBLY_ID}`);
+		error(500, `Cannot find the cabinet: ${id}`);
 	}
+	return cabinetAssembly;
+}
 
-	const politicians = await fetchPoliticians();
-	const parties = await fetchParties();
-	const promises = await fetchPromises();
+function findPrimeMinister(cabinetMembers: AssemblyMember[]) {
+	const primeMinister = cabinetMembers.find((m) => m.assemblyRole?.role === 'นายกรัฐมนตรี');
+	if (!primeMinister) {
+		error(500, `Cannot find the prime minister in the given cabinet`);
+	}
+	return primeMinister;
+}
 
-	const cabinetMembers = getAssemblyMembers(cabinetAssembly, politicians).filter(
+function getCainetMembers(cabinetAssembly: Assembly, politicians: Politician[]) {
+	return getAssemblyMembers(cabinetAssembly, politicians).filter(
 		({ assemblyRole }) =>
 			!assemblyRole?.endedAt ||
 			(cabinetAssembly.endedAt && !dayjs(cabinetAssembly.endedAt).isAfter(assemblyRole.endedAt))
 	);
+}
 
+function getCabinetMemberCountsByParty(cabinetMembers: any[], parties: Party[]) {
```

Comment:
We should avoid using `any`. Following the codes, `cabinetMembers` should have the type `AssemblyMember` (imported from `$lib/datasheets/assembly-member`), Or something is not working here?